components/tools/fibonacciFan.js

```diff
@@ -3,395 +3,192 @@
-
-            lineC = root.append('line')
-                .attr('class', 'fibonacci-fan c')
-                .attr('display', 'none');
-
-            fanArea = root.append('polygon')
-                .attr('class', 'fibonacci-fan area')
-                .attr('display', 'none');
-        };
-
-        function mousemove() {
-
-            if (!active) {
-                return;
-            }
-
-            switch (phase) {
-                case 1: {
-                    locationOrigin = findLocation();
-                    if (locationOrigin) {
-                        fibonacciFan.update();
-                        circleOrigin.attr('display', 'inherit');
-                    }
-                    break;
+            selection.each(function() {
+                var data = this.__data__ || [];
+                if (!data.__fan__) {
+                    data.__fan__ = {};
+                    this.__data__ = data;
                 }
-                case 2: {
-                    locationTarget = findLocation();
-                    if (locationTarget) {
-                        fibonacciFan.update();
-                        circleTarget.attr('display', 'inherit');
-                        lineSource.attr('display', 'inherit');
-                    }
-                    break;
-                }
-                case 3: {
-                    break;
-                }
-            }
-        }
-
-        function mouseclick() {
-
-            if (!active) {
-                return;
-            }
-
-            switch (phase) {
-                case 1: {
-
-                    phase = 2;
-                    break;
-                }
-                case 2: {
-
-                    setFan();
-
-                    phase = 3;
-                    break;
-                }
-                case 3: {
-
-                    clearAll();
-
-                    phase = 1;
-                    break;
-                }
-            }
-        }
-
-        function findLocation() {
-
-            var mouse = [0, 0];
-            try {
-                mouse = d3.mouse(target[0][0]);
-            }
-            catch (exception) {
-                // Mouse is elsewhere
-            }
-
-            var xMouse = xScale.invert(mouse[0]),
-                yMouse = yScale.invert(mouse[1]),
-                point = findPoint(xMouse);
-
-            if (point !== null) {
+            });
 
-                var field = findField(yMouse, point);
+            selection.each(function(data) {
 
-                if (field !== null) {
+                var container = d3.select(this)
+                    .style('pointer-events', 'all')
+                    .on('mouseenter.fan', mouseenter);
 
-                    return {point: point, field: field};
+                if (!data.__fan__.overlay) {
+                    container.append('rect')
+                        .style('visibility', 'hidden');
+                    data.__fan__.overlay = true;
```

Comment:
I appreciate this is the same for all tools which have had similar updates, so I'm happy with this for now, but when clearing the data and calling the component it can lead to a lot of unnecessary elements being appended:

![image](https://cloud.githubusercontent.com/assets/2017615/5756569/e158d1aa-9ca8-11e4-80f4-eb9018e2f873.png)

What do you think about using a data join with dummy data instead, like this:

```js
// Data join with a single datum - to create only one overlay
var overlay = container.selectAll('rect.overlay').data([0]);

// Enter
overlay.enter().append('rect')
    .attr('class', 'overlay')
    .style('visibility', 'hidden');

// Update
overlay.attr('x', fan.xScale.value.range()[0])
    .attr('y', fan.yScale.value.range()[1])
    .attr('width', fan.xScale.value.range()[1])
    .attr('height', fan.yScale.value.range()[0]);

// Exit - remove any unnecessary overlays
overlay.exit().remove();
```

It would also mean that we don't have to mess about with data, the opening loop could be removed and something like this could be added to the main loop:

```js
this.__data__ = this.__data__ || [];
```